src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -442,7 +442,16 @@ private PoolEntry createPoolEntry()
 
          final long maxLifetime = config.getMaxLifetime();
          if (maxLifetime > 0) {
-            final long variance = maxLifetime > 60_000 ? ThreadLocalRandom.current().nextLong(10_000) : 0;
+        	 long inHours = TimeUnit.MILLISECONDS.toHours(maxLifetime);
+        	 
+        	 // Every hour will add a 10 second random factor for the maxlifetime
+        	 long randomIntervalInMillis = inHours * 10_000;
+        	 
+        	 // Support the up to 10 seconds spread out when the lifetime is less than 1 hour
+        	 if (inHours == 0) {
```

Comment:
as maxLifetime can not be less than 30sec, better change would be just:
final long variance = Math.min( ThreadLocalRandom.current().nextLong( maxLifetime / 1000 ), maxPoolSize * 1000);